projects/rocprofiler-systems/tests/generate_papi_nic_events.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# MIT License
+#
+# Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# This script gets a list of default NICs from ip command
+# and generates a list of PAPI events, 4 for each NIC.
+# For example, if the NIC is enp7s0, the PAPI events are
+# net:::enp7s0:tx:byte net:::enp7s0:rx:byte net:::enp7s0:tx:packet net:::enp7s0:rx:packet
+
+nic_list=$(ip r | awk '/default/{print $5}')
+
+event_list=
+for nic in $nic_list
+do
+  event_list="$event_list net:::$nic:tx:byte net:::$nic:rx:byte net:::$nic:tx:packet net:::$nic:rx:packet"
+done
```

Comment:
May be overkill. Just trying to get rid of that leading whitespace.

```suggestion
events=() 

for nic in $nic_list; do  
  events+=("net:::${nic}:tx:byte" "net:::${nic}:rx:byte" "net:::${nic}:tx:packet" "net:::${nic}:rx:packet")  
done  

event_list="${events[*]}" 
```